pkg/source/docker.go

```diff
@@ -29,6 +29,10 @@ func (ds *DockerSource) Ref() meta.OCIImageRef {
 	return ds.imageRef
 }
 
+func (ds *DockerSource) SetRef(imageRef meta.OCIImageRef) {
+	ds.imageRef = imageRef
+}
```

Comment:
`SetRef` is required to read the correct imageRef in `DockerSource.Reader` when a single image tar file contains multiple images. DockerSource importing multiple images require this option to set the imageRef of the DockerSource when the images are read.